libopae/src/sysobject.c

```diff
@@ -98,18 +101,42 @@ fpga_result __FPGA_API__ fpgaReadObjectBytes(fpga_token token, const char *key,
 					     uint8_t *buffer, size_t offset,
 					     size_t *len)
 {
+	int fd = -1;
+	char objpath[SYSFS_PATH_MAX];
+	ssize_t count = 0;
+	fpga_result res = FPGA_EXCEPTION;
+
 	NULL_CHECK(token);
 	NULL_CHECK(key);
 	NULL_CHECK(buffer);
 	NULL_CHECK(len);
-	(void)offset;
 
-	char objpath[SYSFS_PATH_MAX];
-	fpga_result res = cat_token_sysfs_path(objpath, token, key);
+	res = cat_token_sysfs_path(objpath, token, key);
 	if (res) {
 		return res;
 	}
 
+	fd = open(objpath, O_RDONLY);
+	if (fd < 0) {
+		return FPGA_NOT_FOUND;
+	}
+
+	count = pread(fd, buffer, *len, offset);
```

Comment:
Why not loop here if the call fails and errno == EINTR?  man pread/pwrite says that the fn's can fail and set any error code that read/write can set.